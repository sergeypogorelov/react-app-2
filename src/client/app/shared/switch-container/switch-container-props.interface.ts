import { SwitchProps } from './switch/switch-props.interface';

export interface SwitchContainerProps {
  switchProps: SwitchProps;
}
