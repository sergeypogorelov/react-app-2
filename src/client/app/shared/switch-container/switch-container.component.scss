@import '../../../styles/variables';

.switch-container {
  &-label {
    margin-right: 20px;
  }
  opacity: 0.8;
  font-family: $f-medium;
  font-size: 16px;
}
