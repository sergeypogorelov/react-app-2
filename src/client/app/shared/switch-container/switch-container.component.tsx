import './switch-container.component.scss';

import React, { FunctionComponent } from 'react';

import { Switch } from './switch/switch.component';

import { SwitchContainerProps } from './switch-container-props.interface';

export const SwitchContainer: FunctionComponent<SwitchContainerProps> = ({
  switchProps,
  children,
}) => {
  return (
    <div className="switch-container d-flex align-items-center">
      <label className="switch-container-label my-0">{children}</label>
      <Switch {...switchProps} />
    </div>
  );
};
