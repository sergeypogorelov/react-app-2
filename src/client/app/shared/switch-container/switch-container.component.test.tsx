import React from 'react';
import renderer from 'react-test-renderer';

import { SwitchContainer } from './switch-container.component';
import { SwitchContainerProps } from './switch-container-props.interface';

test('SwitchContainer component should render correctly', () => {
  const props: SwitchContainerProps = {
    switchProps: {
      left: {
        label: 'left label',
        value: 'left value',
      },
      right: {
        label: 'right label',
        value: 'right value',
      },
      currentValue: 'left value',
      onChange: jest.fn(),
    },
  };

  const tree = renderer
    .create(<SwitchContainer {...props}>children</SwitchContainer>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
