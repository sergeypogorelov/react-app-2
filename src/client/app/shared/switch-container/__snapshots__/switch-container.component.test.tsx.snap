// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchContainer component should render correctly 1`] = `
<div
  className="switch-container d-flex align-items-center"
>
  <label
    className="switch-container-label my-0"
  >
    children
  </label>
  <div
    className="switch d-flex"
  >
    <button
      className="switch-item switch-item-left active"
      onClick={[Function]}
      type="button"
    >
      left label
    </button>
    <button
      className="switch-item switch-item-right"
      onClick={[Function]}
      type="button"
    >
      right label
    </button>
  </div>
</div>
`;
